### PR TITLE
Blog post: "What is Kubernetes? And how you can save 50% on your cloud bill"

### PR DESCRIPTION
**Pull request by danielepolencic**
Preview: https://deploy-preview-107--learnk8s.netlify.com/blog/what-is-kubernetes

### CODE REVIEW

File: _blog/why-kube/index.md [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+
+They also have to emulate hardware and come with the extra baggage of an operating system.
+
+Linux containers, on the other hand, are merely processes running on your host.
+
+Indeed, for the same operating system and server, you could have dozens of containers running on that host.
+
+And despite living on the same computer, processes running in containers can't see each other.
+
+Applications running inside containers are entirely isolated and can't tell the difference between a virtual machine and a container.
```
[the remark this answers is not in the document]
Comment (danielepolencic) [reply _blog/why-kube/index.md]: I take your point, Docker provides a different kind of isolation compared to VMs. Do you think the word _entirely_ is misleading? Or are you referring to the last sentence?

File: _blog/what-is-kubernetes/index.md [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+
+You put a lot of effort into tuning the system only to realise that the application changed again and you have to start from scratch.
+
+So you end up taking the only sensible choice: selecting a small, medium and large profile for virtual machines and using them for all workloads.
+
+You know you have to live with wasting hundreds of megabytes of RAM and plenty of CPU cycles.
+
+{% include_relative waste.html %}
+
+If it makes you feel any better, there're plenty of companies suffering from similar inefficiencies.
```
Comment (keithmifsud) [inline _blog/what-is-kubernetes/index.md]: I'm not using Grammarly but this sounds off _there're_. I don't think this abbreviation is grammatically correct.

File: _blog/what-is-kubernetes/index.md [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+
+Ideally, you should archive all of the parts necessary to run your component as a single bundle.
+
+No more getting lost in chasing dependencies just before a release.
+
+_Yes, easier said than done._
+
+Or maybe not.
+
+## Borrowing containers from the shipping industry
```
Comment (keithmifsud) [inline _blog/what-is-kubernetes/index.md]: This is really an excellent explanation/comparison.

File: _blog/what-is-kubernetes/index.md [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+
+That's great news!
+
+Linux containers are like virtual machines, but more efficient.
+
+_But what are those Linux containers made of, anyway?_
+
+## Linux containers are isolated processes with benefits
+
+The magic of containers comes from two feature in the Linux kernel: control groups and namespaces.
```
Comment (keithmifsud) [inline _blog/what-is-kubernetes/index.md]: _features_ plural.

File: _blog/what-is-kubernetes/index.md [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+
+Namespaces, on the other hand, are in charge of isolating the process and limiting what it can see.
+
+The component can only see the network packets that are directly related to it.
+
+It won't be able to see all of the network packets flowing through the network adapter.
+
+Control groups and namespaces are low-level primitives.
+
+With time developers created more and more layers of abstractions to make it easier to control those kernel features.
```
Comment (keithmifsud) [inline _blog/what-is-kubernetes/index.md]: Add a comma after "With time".

File: _blog/what-is-kubernetes/index.md [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+
+## Final notes
+
+Kubernetes and containers are here to stay.
+
+With backing from companies such as Google, Microsoft, Red Hat, Pivotal, Oracle, IBM, and many more it's hard to believe that it won't catch on.
+
+Many companies are getting a head start on Kubernetes and joining the revolution.
+
+Not just startups and SMEs, but big corporations such as banks, financial institutions and insurance companies are betting on containers and Kubernetes to be the future.
```
Comment (keithmifsud) [inline _blog/what-is-kubernetes/index.md]: Suggestion: include an industry type around embedded systems 😉

File: _blog/what-is-kubernetes/index.md [hunk 7]
```diff
@@ -0,0 +1,469 @@
+---
+layout: post
+title: What is Kubernetes? And how you can save 50% on your cloud bill
```
[the remark this answers is not in the document]
Comment (danielepolencic) [reply _blog/what-is-kubernetes/index.md]: Oops.

File: _blog/what-is-kubernetes/index.md [hunk 8]
```diff
@@ -0,0 +1,469 @@
...
+
+In a large organisation, managing hundreds of applications is demanding, but still doable.
+
+You have several teams dedicated to developing, packaging and releasing applications.
+
+**Developing services out of smaller components, on the other hand, introduces a different challenge.**
+
+When for every application, you can refactor the same apps in a collection of four components, you have at least four times more apps to develop, package and release.
+
+{% include_relative monolith.html %}
```
[the remark this answers is not in the document]
Comment (keithmifsud) [reply _blog/what-is-kubernetes/index.md]: So worker is also a bit off.

File: _blog/what-is-kubernetes/index.md [hunk 9]
```diff
@@ -0,0 +1,469 @@
...
+
+Even if your component can run with only 1GB of memory.
+
+_Yes, you could optimise in the future._
+
+But let's be honest: **it's like changing tyres while driving**.
+
+You put a lot of effort into tuning the system only to realise that the application changed again and you have to start from scratch.
+
+So you end up taking the only sensible choice: selecting a small, medium and large profile for virtual machines and using them for all workloads.
```
[the remark this answers is not in the document]
Comment (danielepolencic) [reply _blog/what-is-kubernetes/index.md]: I understand the confusion. Small, medium and large profiles are 3 profiles. That's why I have "them".

[the remark this answers is not in the document]
Comment (keithmifsud) [reply _blog/what-is-kubernetes/index.md]: 👍

File: _blog/what-is-kubernetes/index.md (hunk 2 above)
Comment (danielepolencic) [reply _blog/what-is-kubernetes/index.md]: I can have `there are`. Is that what you meant?

Comment (keithmifsud) [reply _blog/what-is-kubernetes/index.md]: Yes, that's what I mean.

File: _blog/what-is-kubernetes/index.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (danielepolencic) [reply _blog/what-is-kubernetes/index.md]: image

[the remark this answers is not in the document]
Comment (keithmifsud) [reply _blog/what-is-kubernetes/index.md]: Look good here.

[the remark this answers is not in the document]
Comment (danielepolencic) [reply _blog/what-is-kubernetes/index.md]: I will reduce the font-size

[the remark this answers is not in the document]
Comment (keithmifsud) [reply _blog/what-is-kubernetes/index.md]: Okies 👍

**Pull request merged**